gwr/kernels.py

```diff
@@ -61,11 +63,22 @@ def __init__(self, data, bandwidth=None, fixed=True, k=None,
             self.k = int(k) + 1
         else:
             self.k = k
-        if points is None:
-            self.dmat = cdist(self.data, self.data)
+        
```

Comment:
The logic here is kind of thick. After stepping it through it, it makes sense, so I don't see any issues for the PR, but we might think about a way simplify it in the future.